materialize-boilerplate/logging.go

```diff
@@ -10,12 +10,12 @@ import (
 
 const (
 	// How often various the various async loggers will report their status.
-	loggingFrequency = 5 * time.Second
+	loggingFrequency = 5 * time.Minute
```

Comment:
I'll update the comment to state this as well:

> When "extended" logging is enabled, transitions of interest will always be logged immediately, and only after `loggingFrequency` will status updates be logged if nothing else has happened. When extended logging is not 	enabled, there will be a more generic status update logged out at this frequency.

So with extended logged for an OLAP materialization, we will for example see a "started reading store requests" logged when it starts doing that, and then every 5 minutes until it finishes we'll get a "reading store requests in progress" log. For materializations that don't enable extended logging it would be pretty spamming to log out such detail information for every transaction, so they just use a logger than logs a general progress message at the `loggingFrequency`.

Reply:
@williamhbaker is there a requirement that makes us want to supress the instant logs from most connectors? I personally find them very useful in debugging connectors. I understand I can change the log level to debug to achieve this, but for already-running tasks that have had an issue in the past, it is useful to be able to go back in time and see what happened in what sequence 🤔

Reply:
I was looking at the original issue that this pull-request aims to address, and the description by Phil is:
> We've had a number of debugging sessions that suffered from lack of information in the logs. Ideally, we'd have some common logging in materialize-sql to help with this sort of thing.

So I think we want _more information_ in logs, rather than less

Reply:
As a point of clarification, with this change we will indeed be logging more for _all_ materializations, just not quite as much more for some vs. others. The simple logger will give us less information than the extended logger, but the simple logger does give us more information than we currently get from materializations that will use it.

With this PR, OLAP materializations that typically run slow transactions will log _a lot_ more, bringing in nearly all of the debug level logs that were previously in `RunTransactions` and logging those at info level, as well as status updates every `loggingFrequency` when it they are in the middle of doing some long running operation like reading a lot of Store requests or running a long transaction.

Other materializations that may run multiple transactions per second will also log more than they did before, but not as much more by default as OLAP materializations since that would be pretty spammy in most cases. Materializations that do not set `ExtendedLogging` and are logging at info level will now log the additional status updates every 5 minutes per `loggingFrequency`, which will describe what they are doing to some extent. Admittedly this is not as sophisticated as I could imagine it being, but it will indicate (among other things) when they have been waiting for documents from the runtime for a long time, or are waiting for a long-running possibly stuck commit for a long time, which are a couple of things we've previously seen.

The specific scenario that the original issue was opened for was a case of a `materialize-postgres` task that we suspect got hung up for a long time when opening and installing its checkpoint (see [thread](https://estuaryworkspace.slack.com/archives/C03Q2NRFKDL/p1724932261968259)). Currently for pretty much all materializations in this case the last thing they will log is a line like `opening database` and then nothing else - and for most non OLAP materializations, this is indeed the last thing they will ever log no matter what, including `materialize-postgres`.  This PR includes some logging additions directly targeted at this case in [this commit](https://github.com/estuary/connectors/pull/2019/commits/dfd0860014ade1272b6449479321e66223649dc2), in addition to the other logging it is adding.

Reply:
Okay I think it is easier if we see this in action and then if we find it doesn't quite satisfy troubleshooting needs we will add more to it. It is hard for me to imagine what the logs look like now. But this is a significant work that will pave the way for better logging, thanks!